src/providers/ldap/sdap_async_nested_groups.c

```diff
@@ -2368,18 +2412,17 @@ sdap_nested_group_deref_direct_process(struct tevent_req *subreq)
         }
 
         if (!member_found) {
-            /* Append newly found member to member list.
+            /* Store newly found member to append to member list later.
              * Changes in state->members will propagate into sysdb_attrs of
              * the group. */
-            members->values[members->num_values].data =
-                    (uint8_t *)talloc_strdup(members->values, orig_dn);
-            if (members->values[members->num_values].data == NULL) {
-                ret = ENOMEM;
-                goto done;
+            key.type = HASH_KEY_CONST_STRING;
+            key.c_str = orig_dn;
+            value.type = HASH_VALUE_UNDEF;
+            ret = hash_enter(new_members, &key, &value);
```

Comment:
Here is a caveat: previously new candidate was compared with all existing members (including newly found so far) using `strcasecmp()`.
With this change 2 DNs (in `entries`) that differ only in case, will be treated as different.
Maybe it requires some changes, but I'm not sure I understand why `strcasecmp()` is used.

Reply:
> I'm not sure I understand why `strcasecmp()` is used.

Hi @sumit-bose, do you understand why `strcasecmp()` is used here?
Is it really possible ldap search returns DNs that differ only in case?
If yes then why one of DNs will be arbitrarily ignored in members list?